src/Authentication/JWTAuthenticator.php

```diff
@@ -25,201 +34,252 @@
+    {
+        $signerKey = $this->getEnv('JWT_SIGNER_KEY');
+        $signerPath = PathResolver::resolve($signerKey);
+        if ($signerPath) {
+            $password = $this->getEnv('JWT_KEY_PASSWORD', null);
+            return new Key('file://' . $signerPath, $password);
         }
+        return new Key($signerKey);
     }
 
     /**
-     * Setup the keys this has to be done on the spot for if the signer changes between validation cycles
+     * Get public key
+     *
+     * @return Key
+     * @throws LogicException
      */
-    private function setKeys()
+    private function getPublicKey(): Key
     {
         $signerKey = Environment::getEnv('JWT_SIGNER_KEY');
+        $signerPath = PathResolver::resolve($signerKey);
         // If it's a private key, we also need a public key for validation!
-        if (file_exists($signerKey)) {
-            $this->signer = new RsaSha256();
-            $password = Environment::getEnv('JWT_KEY_PASSWORD');
-            $this->privateKey = new Key('file://' . $signerKey, $password ?: null);
-            // We're having an RSA signed key instead of a string
-            $this->publicKey = new Key('file://' . Environment::getEnv('JWT_PUBLIC_KEY'));
-        } else {
-            $this->signer = new Sha256();
-            $this->privateKey = $signerKey;
-            $this->publicKey = $signerKey;
+        if (empty($signerPath)) {
+            return new Key($signerKey);
         }
+
+        // Ensure public key exists
+        $publicKey = Environment::getEnv('JWT_PUBLIC_KEY');
+        $publicPath = PathResolver::resolve($publicKey);
+        if (empty($publicPath)) {
+            throw new LogicException('JWT_PUBLIC_KEY path does not exist');
+        }
+        return new Key('file://' . $publicPath);
     }
 
     /**
      * JWT is stateless, therefore, we don't support anything but login
      *
      * @return int
      */
-    public function supportedServices()
+    public function supportedServices(): int
     {
-        return Authenticator::LOGIN | Authenticator::CMS_LOGIN;
+        return Authenticator::LOGIN;
     }
 
     /**
-     * @param array $data
-     * @param HTTPRequest $request
+     * @param array                 $data
+     * @param HTTPRequest           $request
      * @param ValidationResult|null $result
      * @return Member|null
      * @throws OutOfBoundsException
      * @throws BadMethodCallException
+     * @throws Exception
      */
-    public function authenticate(array $data, HTTPRequest $request, ValidationResult &$result = null)
+    public function authenticate(array $data, HTTPRequest $request, ValidationResult &$result = null): ?Member
     {
         if (!$result) {
             $result = new ValidationResult();
         }
         $token = $data['token'];
 
-        return $this->validateToken($token, $request, $result);
+        /** @var JWTRecord $record */
+        list($record, $status) = $this->validateToken($token, $request);
```

Comment:
I so much hate and love the `list()` record! :D I use it a lot but I still feel dirty for doing it.

Reply:
Wish PHP had object destructuring hah.